pystiche_papers/sanakoyeu_et_al_2018/_data.py

```diff
@@ -0,0 +1,129 @@
+from typing import Any, Dict, Optional, Sized, cast
+
+import torch
+from torch.utils.data import DataLoader, Dataset, Sampler
+
+from pystiche.data import ImageFolderDataset
+from pystiche.image.transforms import ComposedTransform, Transform, ValidRandomCrop
+from pystiche.image.transforms import functional as F
+from pystiche.image.utils import extract_image_size
+
+from ..data.utils import FiniteCycleBatchSampler
+from ..utils import OptionalGrayscaleToFakegrayscale
+
+
+def image_transform(edge_size: int = 768,) -> ComposedTransform:
+    class OptionalRescale(Transform):
```

Comment:
I think something along the lines `BoundedRescale` would be a better fit to describe what is going on. It is not ideal, but I can't think of something better right now. `OptionalRescale` conveys that the rescaling is always optional while `BoundedRescale` conveys that the rescaling within bounds always happens. Ideally, we need something that combines these two.

Reply:
What about `ClampSize`?